test/extended/util/framework.go

```diff
@@ -318,12 +398,27 @@ func WaitForADeployment(client kclient.ReplicationControllerInterface, name stri
 		if err != nil {
 			return err, false
 		}
+		// multiple deployments are conceivable; so we look to see how the latest depoy does
+		lastMatched := false
+		var lastErr error = nil
+		var lastRC string
 		for _, rc := range list.Items {
-			err, matched := okOrFailed(&rc)
-			if matched {
-				return err, false
```

Comment:
i'm unclear on what happened here.  previously this code looped through all the deployments, breaking/returning as soon as it found one that was either complete or failed (ie okOrFailed returned matched==true).  That certainly seems wrong, but rather than checking okOrFailed on every deployment, would it make more sense for the new code to just find the most recent deployment and call okOrFailed on that one?